parser.rst

```diff
@@ -0,0 +1,901 @@
+
+    ./PCbuild/build.bat --regen
+
+The generated parser file is located at :file:`Parser/parser.c`.
+
+How to regenerate the meta-parser
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The meta-grammar (the grammar that describes the grammar for the grammar files
+themselves) is located at :file:`Tools/peg_generator/pegen/metagrammar.gram`.
+Although is very unlikely that you ever need to modify it, if you make any modifications
+to this file (in order to implement new Pegen features) you will need to regenerate
+the meta-parser (the parser that parses the grammar files). To do so just execute: ::
+
+    make regen-metaparser
+
+If you are on Windows you can use the Visual Studio project files
+to regenerate the parser or to execute: ::
+
+    ./PCbuild/build.bat --regen
+
+
+Grammatical elements and rules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Pegen has some special grammatical elements and rules:
+
+* Strings with single quotes (') (e.g. ``'class'``) denote KEYWORDS.
+* Strings with double quotes (") (e.g. ``"match"``) denote SOFT KEYWORDS.
+* Upper case names (e.g. ``NAME``) denote tokens in the :file:`Grammar/Tokens` file.
+* Rule names starting with `invalid_` are used for specialized syntax errors.
+
+  - These rules are NOT used in the first pass of the parser.
+  - Only if the first pass fails to parse, a second pass including the invalid
+    rules will be executed.
+  - If the parser fails in the second phase with a generic syntax error, the
+    location of the generic failure of the first pass will be used (this avoids
+    reporting incorrect locations due to the invalid rules).
+  - The order of the alternatives involving invalid rules matter
+    (like any rule in PEG).
+
+Tokenization
+~~~~~~~~~~~~
+
+It is common among PEG parser frameworks that the parser does both the parsing and the tokenization,
+but this does not happen in Pegen. The reason is that the Python language needs a custom tokenizer
+to handle things like indentation boundaries, some special keywords like ``ASYNC`` and ``AWAIT``
+(for compatibility purposes), backtracking errors (such as unclosed parenthesis), dealing with encoding,
+interactive mode and much more. Some of these reasons are also there for historical purposes, and some
+others are useful even today.
+
+The list of tokens (all uppercase names in the grammar) that you can use can be found in the :file:`Grammar/Tokens`
+file. If you change this file to add new tokens, make sure to regenerate the files by executing: ::
+
+    make regen-token
+
+If you are on Windows you can use the Visual Studio project files to regenerate the tokens or to execute: ::
+
+    ./PCbuild/build.bat --regen
+
+How tokens are generated and the rules governing this is completely up to the tokenizer (:file:`Parser/tokenizer.c`)
+and the parser just receives tokens from it.
+
+Memoization
+~~~~~~~~~~~
+
+As described previously, to avoid exponential time complexity in the parser, memoization is used. In the Python
```

Comment:
There seems to be something missing here?

Reply:
Thanks for the catch! "In the Python" should be removed.